### ISSUE TITLE
'ninja scan-build' broken

### ISSUE THREAD

**Issue by rhd** [mesonbuild/meson#1082]
Hi guys,

I was just poking around with `ninja scan-build` and notice it wasn't working.

The first issue is the [howto](https://github.com/mesonbuild/meson/wiki/howtox) isn't correct. It's `ninja scan-build` not`ninja scanbuild`.

Second, it doesn't use the correct directory when it calls actually calls scan-build.

I believe the patch to fix this to be:

```diff
diff --git a/meson/mesonbuild/scripts/scanbuild.py b/meson/mesonbuild/scripts/scanbuild.py
index dd74ce8..c987cd2 100644
--- a/meson/mesonbuild/scripts/scanbuild.py
+++ b/meson/mesonbuild/scripts/scanbuild.py
@@ -19,7 +19,8 @@ import tempfile
 def scanbuild(exename, srcdir, blddir, privdir, logdir, args):
     with tempfile.TemporaryDirectory(dir=privdir) as scandir:
         meson_cmd = [exename] + args
-        build_cmd = [exename, '-o', logdir, 'ninja']
+        build_cmd = [exename, '-o', logdir, 'ninja', '-C', scandir]
         rc = subprocess.call(meson_cmd + [srcdir, scandir])
         if rc != 0:
```

**Comment by nirbheek** [comment]
I've fixed the documentation. Could you please open a PR with that patch? It looks obviously correct.

**Comment by nirbheek** [comment]
Also, it hard-codes `ninja` which is wrong since it can be `ninja-build`. The full path to the ninja returned by `detect_ninja()` should be passed to the script and used here.

**Comment by nirbheek** [comment]
This is fixed now with #1264,
